discord-scripts/invite-management.ts

```diff
@@ -0,0 +1,134 @@
+import { Robot } from "hubot"
+import { Client, TextChannel, VoiceChannel, StageChannel } from "discord.js"
+
+async function manageInvite(
+  discordClient: Client,
+  guildId: string,
+  channelId: string,
+  action: "create" | "delete",
+  maxAge: number | string = 86400,
+  maxUses: number | string = 0,
+): Promise<string | null> {
+  if (!discordClient) {
+    throw new Error("Discord client is not initialized.")
+  }
+
+  const guild = await discordClient.guilds.fetch(guildId)
+  if (!guild) throw new Error("Guild not found.")
+
+  const channel = await guild.channels.cache.get(channelId)
+  if (!channel) throw new Error("Channel not found.")
+
+  const numericMaxAge =
+    typeof maxAge === "string" ? parseInt(maxAge, 10) : maxAge
+  const numericMaxUses =
+    typeof maxUses === "string" ? parseInt(maxUses, 10) : maxUses
+
+  if (Number.isNaN(numericMaxAge) || Number.isNaN(numericMaxUses)) {
+    throw new Error("maxAge and maxUses must be valid numbers.")
+  }
+
+  if (action === "create") {
+    if (
+      channel instanceof TextChannel ||
+      channel instanceof VoiceChannel ||
+      channel instanceof StageChannel
+    ) {
+      const invite = await channel.createInvite({
+        maxAge: numericMaxAge,
+        maxUses: numericMaxUses,
+        unique: true,
+      })
+      return invite.url
+    }
+  }
+  if (action === "delete") {
+    throw new Error(
+      "Delete action is not supported directly through this function.",
+    )
+  }
+  return null
+}
+
+export default async function sendInvite(discordClient: Client, robot: Robot) {
+  const { application } = discordClient
+
+  if (application) {
+    discordClient.on("messageCreate", async (message) => {
+      if (message.author.bot || !message.content.startsWith("!create-invite"))
+        return
+
+      if (!message.guild) return
+
+      const channelId = message.channel.id
+      const guildId = message.guild.id
+      const maxAge = 86400 // 1 day
+      const maxUses = 10
+
+      try {
+        const inviteUrl = await manageInvite(
+          discordClient,
+          guildId,
+          channelId,
+          "create",
+          maxAge,
+          maxUses,
+        )
+
+        if (inviteUrl) {
+          robot.logger.info("New invite URL:", inviteUrl)
+          message.channel.send(
+            `Here is your invite link: ${inviteUrl}\n` +
+              `This invite expires in ${
+                maxAge / 3600
+              } hours and has a maximum of ${maxUses} uses.`,
+          )
+        } else {
+          message.channel.send("Unable to create an invite link.")
+        }
+      } catch (error) {
+        message.channel.send(`An error occurred: ${error}`)
+      }
+    })
```

Comment:
Let's move this out into a proper Discord command.

Note that even if we wanted to do a plain-text message, we should use the Hubot `robot.hear` functionality, as with e.g. https://github.com/thesis/valkyrie/blob/main/discord-scripts/help-me-archive.ts#L21 .

Reply:
@Shadowfiend I've got this setup now in the latest commit with new command `/create-invite` with the same functionality. I also added two parameters to the command `max-age` and `max-uses` that optionally let you specify the maxAge (in hours) or maxUses (number of uses) to the invite. By default it will be set to 24 hours and 10 invites uses. Does that seem like a good feature or not needed at the moment?

Reply:
I would drop it because it'll make things more complicated for folks.

Reply:
That said—I would use a default time range of 7 days and a default max uses of 10 as you've already done.

If we then see there are issues with this, we can revisit.